core/src/main/java/org/elasticsearch/index/query/TemplateQueryParser.java

```diff
@@ -80,9 +80,9 @@ public Query parse(QueryParseContext parseContext) throws IOException {
         BytesReference querySource = (BytesReference) executable.run();
 
         try (XContentParser qSourceParser = XContentFactory.xContent(querySource).createParser(querySource)) {
-            final QueryParseContext context = new QueryParseContext(parseContext.index(), parseContext.indexQueryParserService());
+            final QueryCreationContext context = new QueryCreationContext(parseContext.context().index(), parseContext.context().indexQueryParserService());
             context.reset(qSourceParser);
-            return context.parseInnerQuery();
+            return context.parseContext().parseInnerQuery();
```

Comment:
not sure why we go through the creation of QueryCreationContext to then retrieve the QueryParseContext, when we have the QueryParseContext in the first place